File: skills/jazzqi/xiaomi-mimo-tts/scripts/smart/mimo_tts_smart.js
#!/usr/bin/env node
// Simple heuristic smart wrapper for Node
const { spawnSync } = require('child_process');
const args = process.argv.slice(2);
if (!args[0]) { console.error('Usage: mimo_tts_smart.js "TEXT" [OUTPUT]'); process.exit(2); }
const text=args[0];
const output=args[1] && !args[1].startsWith('--')?args[1]:`${process.cwd()}/output.mock.ogg`;
// very small heuristic
let style='';
if (/诗|床前|李白/.test(text)) style='温柔';
else if (/笑话|哈哈|笑/.test(text)) style='开心';
else if (/晚安|宝宝/.test(text)) style='温柔';
else if (/唱|歌/.test(text)) style='唱歌';
else if (/东北|老铁|咋/.test(text)) style='东北话';

const node = spawnSync('node', [__dirname+'/../base/mimo_tts.js', `<style>${style}</style>${text}`, output], { stdio: 'inherit' });
process.exit(node.status);

File: skills/jazzqi/xiaomi-mimo-tts/scripts/smart/mimo_tts_smart.py
#!/usr/bin/env python3
"""Simple heuristic smart wrapper for Python
"""
import sys,subprocess,os
if len(sys.argv)<2:
    print('Usage: mimo_tts_smart.py "TEXT" [OUTPUT]')
    sys.exit(2)
text=sys.argv[1]
output=sys.argv[2] if len(sys.argv)>2 and not sys.argv[2].startswith('--') else os.path.join(os.getcwd(),'output.mock.ogg')
style=''
if any(k in text for k in ['诗','床前','李白']): style='温柔'
elif any(k in text for k in ['笑话','哈哈','笑']): style='开心'
elif any(k in text for k in ['晚安','宝宝']): style='温柔'
elif any(k in text for k in ['唱','歌']): style='唱歌'
elif any(k in text for k in ['东北','老铁','咋']): style='东北话'

# call python base implementation
base=os.path.join(os.path.dirname(__file__),'..','base','mimo_tts.py')
cmd=['python3',base,f'<style>{style}</style>'+text,output]
res=subprocess.run(cmd)
sys.exit(res.returncode)
